src/xmipp/libraries/reconstruction/angular_distance.cpp

```diff
@@ -573,3 +625,114 @@ void ProgAngularDistance::computeWeights()
     DF2weighted.removeDisabled();
     DF2weighted.write(fn_out+"_weights.xmd");
 }
+
+void ProgAngularDistance::euler2quat(   double rot, double tilt, double psi,
+                                        double q[4] )
+{
+    // Convert to radians and divide by 2
+    const double ai = -DEG2RAD(rot) / 2;
+    const double aj = DEG2RAD(tilt) / 2;
+    const double ak = -DEG2RAD(psi) / 2;
+
+    // Obtain sin and cos
+    const double    si = std::sin(ai),
+                    ci = std::cos(ai),
+                    sj = std::sin(aj),
+                    cj = std::cos(aj),
+                    sk = std::sin(ak),
+                    ck = std::cos(ak);
+
+    const double    cc = ci * ck,
+                    cs = ci * sk,
+                    sc = si * ck,
+                    ss = si * sk;
+
+    // Compute the quaternion values
+    // WXYZ
+    q[0] = +cj * (cc - ss);
+    q[1] = +sj * (cs - sc);
+    q[2] = -sj * (cc + ss);
+    q[3] = +cj * (cs + sc);
+}
+
+void ProgAngularDistance::quat2Euler(   const double q[4],
```

Comment:
Use "std::array" or "std::vector" instead of a C-style array.

Reply:
Yes, it is a better idea to use std::array